src/PaymentGateways/Gateways/TestGateway/TestGatewayOffsite.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\TestGateway;
+
+use Give\Framework\PaymentGateways\Commands\PaymentComplete;
+use Give\Framework\PaymentGateways\Commands\RedirectOffsite;
+use Give\Framework\PaymentGateways\Contracts\OffsiteGatewayInterface;
+use Give\Framework\PaymentGateways\PaymentGateway;
+use Give\Framework\PaymentGateways\Traits\OffsiteGateway;
+use Give\Helpers\Form\Utils as FormUtils;
+use Give\PaymentGateways\DataTransferObjects\GatewayPaymentData;
+use Give\PaymentGateways\Gateways\TestGateway\Views\LegacyFormFieldMarkup;
+
+/**
+ * Class TestGatewayOffsite
+ * @unreleased
+ */
+class TestGatewayOffsite extends PaymentGateway implements OffsiteGatewayInterface
+{
+    use OffsiteGateway;
+
+    /**
+     * @inheritDoc
+     */
+    public static function id()
+    {
+        return 'test-gateway-offsite';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getId()
+    {
+        return self::id();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getName()
+    {
+        return __('Test Gateway Offsite', 'give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPaymentMethodLabel()
+    {
+        return __('Test Gateway Offsite', 'give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getLegacyFormFieldMarkup($formId)
+    {
+        if (FormUtils::isLegacyForm($formId)) {
+            return false;
+        }
+
+        /** @var LegacyFormFieldMarkup $legacyFormFieldMarkup */
+        $legacyFormFieldMarkup = give(LegacyFormFieldMarkup::class);
+
+        return $legacyFormFieldMarkup();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function createPayment(GatewayPaymentData $paymentData)
+    {
+        $transactionId = "test-gateway-transaction-id-{$paymentData->paymentId}";
```

Comment:
I believe the transaction is generated offsite, so it's available in `returnFromOffsiteRedirect`, not here.

Reply:
Yes, the `transactionId` would likely come after returning but I was just trying to show both an example of adding args to the return url and how to receive data within `returnFromOffsiteRedirect`.  I'm open to using another example but needed something tangible like the `transactionId` to use for this flow example.

Reply:
That's fine I guess, but the `paymentId` wouldn't be available in the returning DTO.

Reply:
I'm sending the `paymentId` in the query params for the return url so it's available when the route resolves.